PLNGatewayPlugin.inc.php

```diff
@@ -32,6 +32,8 @@ class PLNGatewayPlugin extends GatewayPlugin {
 	 * @param $parentPluginName string
 	 */
 	public function __construct($parentPluginName) {
+		// Disable the login restriction for this handler, the PKP PN needs access to this URL
+		$this->getRequest()->getContext()->setData('restrictSiteAccess', 0);
```

Comment:
At least for the `/main`, I was thinking about adding a hook into the [AuthorizationDecisionManager::decide()](https://github.com/pkp/pkp-lib/blob/main/classes/security/authorization/AuthorizationDecisionManager.inc.php#L135).

Reply:
I think a hook in `lib/pkp/classes/security/authorization/RestrictedSiteAccessPolicy.inc.php` in `_getLoginExemptions` might be more targeted. Actually this is a very niche issue so I'd be OK requiring an OJS upgrade to work around it; it could go into the next `stable-3_3_0` release. I'd prefer that to a lot of contortions here that might have unwanted side-effects.

Reply:
Ok! I'll add the hooks 👌